components/fxa-client/src/device.rs

```diff
@@ -53,6 +53,9 @@ impl FirefoxAccount {
                 }
             }
         }
+        // Remember what capabilities we've registered, so we don't register the same ones again.
+        // We write this to internal state before we've actually written the new device record,
+        // but the state-persistence callback will only flush it to disk *after* writing to the server.
         self.state.device_capabilities = capabilities_set;
```

Comment:
If the network request in `ensure_capabilities` fails, this will still be set and we won't try to re-register the capabilities again won't we?

Reply:
Hrm...it's fragile, for sure. My previous reasoning was that the bindings will [only call `tryPersistState` if the network call succeeds](https://github.com/mozilla/application-services/blob/32453768e64fdbbb9d03acb34c27df04d46bd141/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt#L551), meaning we'll retry on next start. But that won't help us if we retry again from the same live account object.

What do you think about adding an explicit `device_capabilities.clear()` in the case this network request fails?

Reply:
This sounds like a good solution to me!